fitsio/fitslib.py

```diff
@@ -4607,9 +4627,35 @@ def verify(self):
         if 'value' not in self:
             raise ValueError("each record must have a 'value' field")
 
+    def _convert_nonstring_value(self, value_orig):
+        """
+        things like 6 and 1.25 are converted with ast.literal_value
+
+        Things like 'hello' are stripped of quotes
+        """
+        import ast
+        if value_orig is None:
+            return value_orig
+
+        try:
+            avalue = ast.parse(value_orig).body[0].value
+            if isinstance(avalue,ast.BinOp):
+                # this is probably a string that happens to look like
+                # a binary operation, e.g. '25-3'
+                value = value_orig
+            else:
+                value = ast.literal_eval(value_orig)
+        except:
+            value = self._convert_string(value_orig)
+
+        if isinstance(value,int) and '_' in value_orig:
+            value = value_orig
+
+        return value
+
```

Comment:
We removed this code because of another bug. I guess we have a test for that bug already so this is fine?

Reply:
this is only used for the specific case we have here, it should be fine.